src/branch/rename.rs

```diff
@@ -1,14 +1,19 @@
+use crate::branch::connections::get_connection_to_modify;
 use crate::branch::context::Context;
 use crate::branch::option::Rename;
+use crate::options::Options;
 use crate::connect::Connection;
 use crate::print;
 
 pub async fn main(
     options: &Rename,
-    _context: &Context,
+    context: &Context,
     connection: &mut Connection,
+    cli_opts: &Options,
 ) -> anyhow::Result<()> {
-    let status = connection
+    let mut modify_connection = get_connection_to_modify(&context.branch, &cli_opts, connection).await?;
```

Comment:
In the case when current connection is not on the source branch, `get_connection_to_modify` should not open any new connections or even do any new roundtrips.

So what we need is something like this:

```
fn get_connection_to_modify(branch, ..., connection) -> ... {
    if connection.branch != branch {
        return branch;
    }
    ...
}
```